File: shared/clang-wrapper/src/main/java/io/ygdrasil/koreos/clang/ClangMemoryException.java
// SPDX-License-Identifier: MIT
package io.ygdrasil.koreos.clang;

/**
 * Exception thrown for memory-related errors.
 */
public class ClangMemoryException extends ClangException {
    public ClangMemoryException(String message) {
        super(message);
    }

    public ClangMemoryException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: shared/clang-wrapper/src/main/kotlin/io/ygdrasil/koreos/clang/ClangMemoryException.kt
// SPDX-License-Identifier: MIT
package io.ygdrasil.koreos.clang

/**
 * Exception thrown for memory-related errors.
 */
class ClangMemoryException @JvmOverloads constructor(
    message: String,
    cause: Throwable? = null
) : ClangException(message, cause)
